### PR TITLE
KNL-1403 BasicEmailService should send all recipient/replyTo headers

### PR DESCRIPTION
**Pull request by ern**

### CODE REVIEW

File: kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java [hunk 1]
```diff
@@ -786,6 +765,23 @@ public void send(String fromStr, String toStr, String subject, String content, S
 		}
 	}
 
+	private InternetAddress[] stringToInternetAddressArray(String addresses) throws AddressException {
```
Comment (jonespm) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java]: This seems like a good change, all for using the library. Though doesn't InternetAddresses.parse do everything this does already?

http://stackoverflow.com/a/13854096/3708872
http://docs.oracle.com/javaee/5/api/javax/mail/internet/InternetAddress.html#parse(java.lang.String)

**Comment by ern** [comment]
Updated with the given input, thanks Matt.

**Comment by buckett** [comment]
The JavaDoc on this should be updated to keep it in sync as this is a change in behaviour.

Why is `replyToStr` now considered required?

**Comment by buckett** [comment]
I think this will break as callers pass in `null` to the replyToStr in a few places in Sakai code. Eg: `org.sakaiproject.tool.resetpass.FormHandler#resetPassClassic`

**Comment by ern** [comment]
I reverted the replyTo header change which allows null.

**Comment by buckett** [comment]
Could you just update the JavaDoc to say it allows multiple headerTo/replyTo values in the string as it used to just accept one. Otherwise it looks good to go.

**Pull request merged**